StringIndexOutOfBoundsException in org.mycore.mets.model.files.File#asElement

<pre>
java.util.concurrent.CompletionException: java.lang.StringIndexOutOfBoundsException: begin -1, end 34, length 34
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1807)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.StringIndexOutOfBoundsException: begin -1, end 34, length 34
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4601)
	at java.base/java.lang.String.substring(String.java:2704)
	at java.base/java.lang.String.substring(String.java:2677)
	at de.uni_jena.thulb.archive.media.MediaService.isMediaSupported(MediaService.java:180)
	at de.uni_jena.thulb.archive.media.events.MediaEventHandler.encodeMediaFile(MediaEventHandler.java:91)
	at de.uni_jena.thulb.archive.media.events.MediaEventHandler.lambda$handlePathCreated$2(MediaEventHandler.java:75)
	at org.mycore.util.concurrent.MCRTransactionableRunnable.run(MCRTransactionableRunnable.java:82)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
</pre>

--- a/src/main/java/org/mycore/mets/model/files/File.java
+++ b/src/main/java/org/mycore/mets/model/files/File.java
@@ -18,8 +18,6 @@
  */
 package org.mycore.mets.model.files;
 
-import java.util.Optional;
-
 import org.jdom2.Element;
 import org.mycore.mets.model.IMetsElement;
 

--- a/src/main/java/org/mycore/mets/model/files/File.java
+++ b/src/main/java/org/mycore/mets/model/files/File.java
@@ -2,21 +2,23 @@
  * $Revision$ $Date$
  * $LastChangedBy$ Copyright 2010 - Thüringer Universitäts- und
  * Landesbibliothek Jena
- * 
+ *
  * Mets-Editor is free software: you can redistribute it and/or modify it under
  * the terms of the GNU General Public License as published by the Free Software
  * Foundation, either version 3 of the License, or (at your option) any later
  * version.
- * 
+ *
  * Mets-Editor is distributed in the hope that it will be useful, but WITHOUT
  * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
  * details.
- * 
+ *
  * You should have received a copy of the GNU General Public License along with
  * Mets-Editor. If not, see http://www.gnu.org/licenses/.
  */
 package org.mycore.mets.model.files;
+
+import java.util.Optional;
 
 import org.jdom2.Element;
 import org.mycore.mets.model.IMetsElement;
@@ -123,13 +125,15 @@ public class File implements IMetsElement {
 
     /*
      * (non-Javadoc)
-     * 
+     *
      * @see org.mycore.mets.model.IMetsElement#asElement()
      */
     public Element asElement() {
         Element file = new Element("file", IMetsElement.METS);
         file.setAttribute("ID", this.getId());
-        file.setAttribute("MIMETYPE", this.getMimeType());
+        if (getMimeType() != null) {
+            file.setAttribute("MIMETYPE", getMimeType());
+        }
         file.addContent(this.getFLocat().asElement());
         return file;
     }

